packages/test/src/FixtureManager.ts

```diff
@@ -0,0 +1,67 @@
+import { MiddlewareAPI, Middleware } from '@rest-hooks/use-enhanced-reducer';
+import { NetworkManager, actionTypes, ReceiveAction } from '@rest-hooks/core';
+import type { Dispatch, Manager } from '@rest-hooks/core';
+
+import { Fixture, actionFromFixture } from './mockState';
+
+/** Resolves requests matching provided fixtures
+ *
+ */
+export default class FixtureManager implements Manager {
+  protected mockResults: { [key: string]: ReceiveAction } = {};
+  protected declare middleware: Middleware;
+
+  constructor(fixtures: Fixture[]) {
+    for (const fixture of fixtures) {
+      const { key, action } = actionFromFixture(fixture);
+      this.mockResults[key] = action;
+    }
+
+    this.middleware = <R extends React.Reducer<any, any>>({
+      dispatch,
+      getState,
+    }: MiddlewareAPI<R>) => {
+      return (next: Dispatch<R>) => (
+        action: React.ReducerAction<R>,
+      ): Promise<void> => {
+        switch (action.type) {
+          case actionTypes.FETCH_TYPE: {
+            const { key } = action.meta;
+
+            if (key in this.mockResults) {
+              dispatch(this.mockResults[key] as any);
+              return Promise.resolve();
```

Comment:
should this be an async function, and this line just be `return`?

Reply:
We dont' use async functions to avoid the huge added code when transpiling with babel to target older browsers since this is a library.

Reply:
ah makes sense.